src/views/AddExpenses.js

```diff
@@ -1,76 +1,100 @@
 import { useState } from 'react';
-import { convertEurosToCents } from '../utils/helpers';
+import { convertEurosToCents, getUKFormattedDate } from '../utils/helpers';
 
-const AddExpenses = ({ addExpense }) => {
+import ConfirmMonthModal from '../components/ConfirmMonthModal';
+
+const AddExpenses = ({ addExpense, selectedMonth, setSelectedMonth }) => {
   const [title, setTitle] = useState('');
   const [date, setDate] = useState('');
   const [amount, setAmount] = useState('');
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [monthToShow, setMonthToShow] = useState(selectedMonth);
 
   const handleSubmit = evt => {
     evt.preventDefault();
     const expense = { title, date, amount: convertEurosToCents(amount) };
+    if (monthToShow !== selectedMonth) {
+      setIsModalOpen(true);
+    }
     addExpense(expense);
     setTitle('');
     setDate('');
     setAmount('');
   };
 
   return (
-    <div className="card h-100">
-      <h3 className="card-header">Add expense</h3>
-      <div className="card-body">
-        <form className="row g-2" onSubmit={handleSubmit}>
-          <div className="col-lg-6">
-            <div className="form-floating">
-              <input
-                className="form-control"
-                type="text"
-                id="name"
-                name="name"
-                required
-                onChange={evt => setTitle(evt.target.value)}
-                value={title}
-                placeholder="Expense Name"
-              />
-              <label htmlFor="name">Expense name</label>
+    <>
+      <div className="card h-100">
+        <h3 className="card-header">Add expense</h3>
+        <div className="card-body">
+          <form className="row g-2" onSubmit={handleSubmit}>
+            <div className="col-lg-6">
+              <div className="form-floating">
+                <input
+                  className="form-control"
+                  type="text"
+                  id="name"
+                  name="name"
+                  required
+                  onChange={evt => setTitle(evt.target.value)}
+                  value={title}
+                  placeholder="Expense Name"
+                />
+                <label htmlFor="name">Expense name</label>
+              </div>
+            </div>
+            <div className="col-lg-6">
+              <div className="form-floating">
+                <input
+                  className="form-control"
+                  type="date"
+                  id="date"
+                  name="date"
+                  required
+                  onChange={evt => {
+                    setDate(evt.target.value);
+                    setMonthToShow(
```

Comment:
Just thinking off the top of my head here: if we are already storing the date picker value in state, do we also need a state variable for a formatted version of it? What if we just formatted `date` on the spot to check if the date picker month is equal to `selectedMonth` at lines 16-17?

Reply:
Not completely sure whether I get your idea exactly, but I think I tried something similar at the beginning ... What I realised is that we need another piece of state to be passed to the modal which needs to persist to render the month in one of the buttons ... `date` is reset in line 21 after we add the expense ... this is why I used `monthToShow`. Did I get it right?

Reply:
Ah okay, that makes sense. Please ignore my comment!

Reply:
😄  no worries! At the beginning I hoped it was enough to use `date` as well.